### DISCUSSION TITLE
PR 11813

### DISCUSSION THREAD

**Discussion by bghira** [openzfs/zfs#11816] [General]
https://github.com/openzfs/zfs/pull/11813

I don't think that naming a pool after a disk is "reasonable"..

There's a link in /dev/poolname/ to /dev/zdXYZ for zvol device paths. if you name a pool after a disk, it could appear as `/dev/sda1/volume1`.

**Comment by mskarbek** [comment]
But you can name pool after disk - both `sda` and `sda1` work "perfectly well". It's only the Solaris-based disk naming convention that is currently filtered. Linked PR is not about naming pools after disk but allowing to name pool in a way that avoids putting in scripts workarounds such as mine:
```
function gen_pool_uuid {
    while true; do
        POOL_UUID=$(cat /proc/sys/kernel/random/uuid)
        if [[ $POOL_UUID =~ ^([a-b]|[d-z]).* ]]; then
            echo $POOL_UUID
            break
        fi
    done
}
```
Yes, I name my pools using UUID. 🤷‍♂️

**Reply by bghira** [reply]
yes, good point. it should be filtered for all platforms.

**Reply by mskarbek** [reply]
Why? How? Just mention in documentation that naming pools after disks is dumb and can hurt you. That's it. If I want to name my pool `sda-something` then why not? Simple filtering based on `sd.*` or `nvme.*` rules will behave similarly to the Solairs-based one preventing me from using such a name.

**Reply by bghira** [reply]
no one reads the documentation.

**Reply by bghira** [reply]
we're presently running restrictions on the `bootfs` property for zpools, across different platforms; making sure your bootloader doesn't get updated is something that can be simply mentioned in the docs, because it's "dumb and can hurt you". i see no reason why we shouldn't filter problematic names across platforms, too. of course, if we shouldn't do one then we shouldn't do the other.
